plugins/request.rb

```diff
@@ -0,0 +1,286 @@
+          self.send(handler_method, opts, opt_parser)
+        else
+          # this should be dead code if parse_args is doing it's job correctly
+          help(opt_parser, "No request handler found for type (#{type.to_s}).")
+        end
+      else
+        if types.include? type
+          help(opt_parser)
+        else
+          help
+        end
+      end
+    end
+
+    def parse_args(args, type = 'http')
+      type.downcase!
+      parse_method = "parse_args_#{type}".to_sym
+      if self.respond_to?(parse_method)
+        self.send(parse_method, args, type)
+      else
+        print_error("Unsupported URI type: #{type}")
+      end
+    end
+
+    # arg parsing for requests of type 'http'
+    def parse_args_https(args = [], type = 'https')
+      # just let http do it
+      parse_args_http(args, type)
+    end
+
+    # arg parsing for requests of type 'http'
+    def parse_args_http(args = [], type = 'http')
+      opt_parser = Rex::Parser::Arguments.new(
+        '-0' => [ false, 'Use HTTP 1.0' ],
+        '-1' => [ false, 'Use TLSv1 (SSL)' ],
+        '-2' => [ false, 'Use SSLv2 (SSL)' ],
+        '-3' => [ false, 'Use SSLv3 (SSL)' ],
+        '-A' => [ true,  'User-Agent to send to server' ],
+        '-d' => [ true,  'HTTP POST data' ],
+        '-G' => [ false, 'Send the -d data with an HTTP GET' ],
+        '-h' => [ false, 'This help text' ],
+        '-H' => [ true,  'Custom header to pass to server' ],
+        '-i' => [ false, 'Include headers in the output' ],
+        '-I' => [ false, 'Show document info only' ],
+        '-o' => [ true,  'Write output to <file> instead of stdout' ],
+        '-u' => [ true,  'Server user and password' ],
+        '-X' => [ true,  'Request method to use' ]
+        #'-x' => [ true,  'Proxy to use, format: [proto://][user:pass@]host[:port]' +
+        #                 '  Proto defaults to http:// and port to 1080'],
+      )
+
+      options = {
+        :headers         => {},
+        :print_body      => true,
+        :print_headers   => false,
+        :ssl_version     => 'Auto',
+        :user_agent      => Rex::Proto::Http::Client::DefaultUserAgent,
+        :version         => '1.1'
+      }
+
+      opt_parser.parse(args) do |opt, idx, val|
+        case opt
+        when '-0'
+          options[:version] = '1.0'
+        when '-1'
+          options[:ssl_version] = 'TLS1'
+        when '-2'
+          options[:ssl_version] = 'SSL2'
+        when '-3'
+          options[:ssl_version] = 'SSL3'
+        when '-A'
+          options[:user_agent] = val
+        when '-d'
+          options[:data] = val
+          options[:method] ||= 'POST'
+        when '-G'
+          options[:method] = 'GET'
+        when HELP_REGEX
+          #help(opt_parser)
+          # guard to prevent further option processing & stymie request handling
+          return [nil, opt_parser]
+        when '-H'
+          name, value = val.split(':', 2)
+          options[:headers][name] = value.strip
```

Comment:
options[:headers][name] = value.to_s.strip if name

might work, tho a caller would have to handle if value.to_s.strip becomes nil